src/scikit_build_core/settings/skbuild_model.py

```diff
@@ -29,6 +30,11 @@ def __dir__() -> List[str]:
     return __all__
 
 
+class SettingsFieldMetadata(TypedDict, total=False):
```

Comment:
I wanted to use TypedDict initially, but somehow I missed that `total` was available in 3.8